AsyncDisplayKit/TextKit/ASTextKitFontSizeAdjuster.mm

```diff
@@ -85,7 +87,7 @@ - (NSUInteger)lineCountForString:(NSAttributedString *)attributedString
   NSLayoutManager *layoutManager = _attributes.layoutManagerCreationBlock ? _attributes.layoutManagerCreationBlock() : [[ASLayoutManager alloc] init];
```

Comment:
These objects, the text storage and the layout manager, are quite expensive to create and even to destroy. They are also required by the renderer. Because many strings will fit without shrinking, it seems quite unfortunate that we must create these objects at least twice.

I don't yet understand all of the mechanics of this awesome system you have put together, so it is possible that we only ever get to this point if we already know that a resize operation is required. However, it would certainly be fantastic and very likely worthwhile if we could either use a text renderer object here for this task, or transfer ownership of these objects after the correct size fit is found so that the drawing operation does not have to re-create them.

![screen shot 2016-03-08 at 3 04 51 am](https://cloud.githubusercontent.com/assets/565251/13632766/32c2b600-e5a0-11e5-9f7d-429d7c5afe73.png)